internal/gatewayapi/resource.go

```diff
@@ -20,41 +24,6 @@ import (
 
 type XdsIRMap map[string]*ir.Xds
```

Comment:
Should we also use list for xdsIR infraIR instead of map

Reply:
thats an internal structure, iirc each resource is unrelated and the map improves lookup

Reply:
https://github.com/envoyproxy/gateway/blob/3e70498b63803d8bd78dcad31883bd7211cda3f8/internal/ir/xds.go#L87 helps deal with out of order equality when comparing ir xds